src/v/iceberg/manifest_entry_values.cc

```diff
@@ -0,0 +1,244 @@
+    ret->fields.emplace_back(
+      long_value(static_cast<int64_t>(file.record_count)));
+    ret->fields.emplace_back(
+      long_value(static_cast<int64_t>(file.file_size_bytes)));
+
+    // TODO: serialize the rest of the optional fields.
+    ret->fields.emplace_back(std::nullopt);
+    ret->fields.emplace_back(std::nullopt);
+    ret->fields.emplace_back(std::nullopt);
+    ret->fields.emplace_back(std::nullopt);
+    ret->fields.emplace_back(std::nullopt);
+    ret->fields.emplace_back(std::nullopt);
+    ret->fields.emplace_back(std::nullopt);
+    ret->fields.emplace_back(std::nullopt);
+    ret->fields.emplace_back(std::nullopt);
+    ret->fields.emplace_back(std::nullopt);
+    return ret;
+}
+
+data_file data_file_from_value(struct_value v) {
+    data_file file;
+    auto& fs = v.fields;
+    if (fs.size() < 10) {
+        throw std::invalid_argument("Expected more values");
+    }
+    file.content_type = content_from_int(
+      get_required_primitive<int_value>(std::move(fs[0])));
+    file.file_path = from_iobuf(
+      get_required_primitive<string_value>(std::move(fs[1])));
+    file.file_format = format_from_str(
+      from_iobuf(get_required_primitive<string_value>(std::move(fs[2]))));
+    file.partition = {get_required_struct(std::move(fs[3]))};
+    file.record_count = get_required_primitive<long_value>(std::move(fs[4]));
+    file.file_size_bytes = get_required_primitive<long_value>(std::move(fs[5]));
+    file.column_sizes = get_counts_map(std::move(fs[6]));
+    file.value_counts = get_counts_map(std::move(fs[7]));
+    file.null_value_counts = get_counts_map(std::move(fs[8]));
+    file.nan_value_counts = get_counts_map(std::move(fs[9]));
+    return file;
+}
+
+} // namespace
+
+struct_value manifest_entry_to_value(const manifest_entry& entry) {
+    struct_value ret;
+    ret.fields.emplace_back(int_value(status_to_int(entry.status)));
+    ret.fields.emplace_back(to_optional_value<long_value>(entry.snapshot_id));
+    ret.fields.emplace_back(
+      to_optional_value<long_value>(entry.sequence_number));
+    ret.fields.emplace_back(
+      to_optional_value<long_value>(entry.file_sequence_number));
+    ret.fields.emplace_back(data_file_to_value(entry.data_file));
+    return ret;
+}
+
+manifest_entry manifest_entry_from_value(struct_value v) {
+    manifest_entry e;
+    auto& fs = v.fields;
+    if (fs.size() < 5) {
+        throw std::invalid_argument("Expected more values");
+    }
+    e.status = status_from_int(
+      get_required_primitive<int_value>(std::move(fs[0])));
+    e.snapshot_id = get_optional_primitive<snapshot_id, long_value>(
+      std::move(fs[1]));
+    e.sequence_number = get_optional_primitive<sequence_number, long_value>(
+      std::move(fs[2]));
+    e.file_sequence_number
+      = get_optional_primitive<file_sequence_number, long_value>(
+        std::move(fs[3]));
+    e.data_file = data_file_from_value(
+      std::move(*get_required_struct(std::move(fs[4]))));
+    return e;
```

Comment:
Mention that the ordering here must match the to_value code, and the manifest entry schema order

Reply:
Done